packages/tkg-clusterclass-aws/bundle/config/upstream/overlay-pod-security-admission.yaml

```diff
@@ -64,71 +64,104 @@ spec:
+                    hostPath: /etc/kubernetes/eventconfig.yaml
+                    mountPath: /etc/kubernetes/eventconfig.yaml
+                  {{- end}}
+            - op: add
+              path: "/spec/template/spec/kubeadmConfigSpec/files/-"
+              valueFrom:
+                template: |-
+                  path: /etc/kubernetes/admission-control-config.yaml
+                  content: |-
+                    {{ $namespace_exemptions := printf "%q, %q" "kube-system" "tkg-system" -}}
+                    {{ $defaultWarnAudit := "baseline" }}
+                    {{- if .podSecurityStandard.exemptions.namespaces -}}
+                      {{ range $namespace := .podSecurityStandard.exemptions.namespaces -}}
+                        {{ $namespace_exemptions = printf "%s, %q" $namespace_exemptions $namespace -}}
+                      {{- end -}}
+                    {{- end -}}
+                    apiVersion: apiserver.config.k8s.io/v1
+                    kind: AdmissionConfiguration
+                    plugins:
+                    {{ if and (not .podSecurityStandard.deactivated) (semverCompare ">= v1.24" .builtin.controlPlane.version) }}
+                    - name: PodSecurity
+                      configuration:
+                        apiVersion: pod-security.admission.config.k8s.io/v1beta1
+                        kind: PodSecurityConfiguration
+                        defaults:
+                          enforce: "{{ if .podSecurityStandard.enforce -}}
+                              {{ .podSecurityStandard.enforce }}
+                            {{- end }}"
+                          enforce-version: "{{ .podSecurityStandard.enforceVersion -}}"
+                          audit: "{{ if .podSecurityStandard.audit -}}
+                              {{ .podSecurityStandard.audit }}
+                            {{- else -}}
+                              {{ $defaultWarnAudit }}
+                            {{- end }}"
+                          audit-version: "{{ .podSecurityStandard.auditVersion -}}"
+                          warn: "{{ if .podSecurityStandard.warn -}}
+                              {{ .podSecurityStandard.warn }}
+                            {{- else -}}
+                              {{ $defaultWarnAudit }}
+                            {{- end }}"
+                          warn-version: "{{ .podSecurityStandard.warnVersion -}}"
+                        exemptions:
+                          usernames: []
+                          runtimeClasses: []
+                          namespaces: [{{ $namespace_exemptions }}]
+                    {{- end }}
+                    {{ if .apiserverEventRateLimitConf }}
+                    - name: EventRateLimit
+                      path: eventconfig.yaml
+                    path: /etc/kubernetes/eventconfig.yaml
+                    content: |-
+                      {{ .apiserverEventRateLimitConf}}
+                    {{- end }}
```

Comment:
also here: needs to be a separate patch, the `add` json operation only allows to add a single entry to the array.
Also toggle it in a separate patch via `enabledIf: {{ .apiserverEventRateLimitConf}}`

or similar.